api/src/main/java/io/druid/data/input/InputRow.java

```diff
@@ -28,8 +30,8 @@
  * implement "schema-less" data ingestion that allows the system to add new dimensions as they appear.
  *
  */
-public interface
-    InputRow extends Row
+@PublicApi
```

Comment:
I was thinking they would all use MapBasedRow, since the current parsers do that, rather than implement their own Row subclasses. We can make it an extension point if you think that makes more sense though.